pages/reference/base-images/base-images.md

```diff
@@ -10,91 +10,90 @@ excerpt: Docker images maintained by {{ $names.company.lower }}
 - [`run`](#run-vs-build) and [`build`](#run-vs-build) variants designed for multistage builds.
 - [cross-build](#building-arm-containers-on-x86-machines) functionality for building ARM containers on x86.
 - Helpful package installer script called `install_packages` inspired by [minideb](https://github.com/bitnami/minideb#why-use-minideb).
 
-
 ## How to Pick a Base Image
 
-<!-- TODO: Add image on how image hierarchy and discuss  -->
-When starting out a project its generally easier to have a "fatter" image which contains a lot of prebuilt dependencies and tools. These images help you get setup faster and work out the requirements for your project. For this reason its recommended to start with `-build` variants and as your project progresses switch to a `-run` variant with some [docker multistage build magic][multistage-build-docs] to slim your deploy image down. In most cases your project can just use a Debian based distribution, which is the default if not specified, but if you know the requirements of your project or prefer specific distros, Ubuntu, Alpine and Fedora images are available. The set of `balenalib` base images follow a simple naming scheme described below, which will help you select a base image for your specific needs.
+<!-- TODO: Add image on how image hierarchy and discuss -->
+When starting out a project, it's generally easier to have a "fatter" image, which contains a lot of prebuilt dependencies and tools. These images help you get setup faster and work out the requirements for your project. For this reason, it's recommended to start with `-build` variants, and as your project progresses, switch to a `-run` variant with some [docker multistage build magic][multistage-build-docs] to slim your deploy image down. In most cases, your project can just use a Debian based distribution, which is the default if not specified, but if you know the requirements of your project or prefer specific distros, Ubuntu, Alpine, and Fedora images are available. The set of `balenalib` base images follow a simple naming scheme described below, which will help you select a base image for your specific needs.
 
 ### How the Image Naming Scheme Works
 
-With over 26000 `balenalib` base images to choose from it can be overwhelming to decide which image and tag is correct for your project. To pick the correct image, it helps to understand how the images are named as that indicates what is installed in the image. In general the naming scheme for the `balenalib` image set follows the pattern below:
+With over 26000 `balenalib` base images to choose from, it can be overwhelming to decide which image and tag are correct for your project. To pick the correct image, it helps to understand how the images are named as that indicates what is installed in the image. In general, the naming scheme for the `balenalib` image set follows the pattern below:
+
 ```
 balenalib/<hw>-<distro>-<lang_stack>:<lang_ver>-<distro_ver>-(build|run)-<yyyymmdd>
```

Comment:
Is it possible to specify `hw`, `distro`, `lang_stack` and `distro_ver` but **not** `lang_ver`?  Or is `lang_ver` compulsory if `lang_stack` is used?

Reply:
Yes as no tags are compulsory. Will default to `latest` if no tag set. Tested via:

```
docker run -it balenalib/raspberrypi3-debian-golang:stretch-run bash
root@ed271cd90107:/go# go version
go version go1.13.4 linux/arm

root@ed271cd90107:/go# cat/etc-os-release
bash: cat/etc-os-release: No such file or directory
root@ed271cd90107:/go# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
```

Reply:
Ah cool! I only asked because it's not clear (from these docs) how the 'punctuation' (the hyphens and colons) would 'collapse' if you specified `hw`, `distro`, `lang_stack` and `distro_ver` but not `lang_ver`'

Could there ever be an edge-case with lang_ver and distro_ver (for the chosen lang_stack and distro) both being numeric, leading to a 'clash' between `balenalib/<hw>-<distro>-<lang_stack>:<lang_ver>` and `balenalib/<hw>-<distro>-<lang_stack>:<distro_ver>` ? (IYSWIM)
Although I'm sure all you clever folks already have a solution for that.

Reply:
Interesting Alpine and Python would fit the bill. I'm guessing there is a simple precedence

`docker run -it balenalib/raspberrypi3-alpine-python:3.7 bash` 

So this gives me Python 3.7 and the latest Alpine (`PRETTY_NAME="Alpine Linux v3.10"`). This makes intuitive sense (to me anyway 😄 ) given the lang ver is the first to be specified. If you wanted Alpine 3.7 and latest Python you'd have to do `docker run -it balenalib/raspberrypi3-alpine-python:latest-3.7 bash` which indeed gives Python 2.7.16 and `PRETTY_NAME="Alpine Linux v3.7"`

Reply:
And I guess rather than trying to explain that in text additional examples would be a good idea.

Reply:
And I guess (for completeness) `balenalib/raspberrypi3-alpine-python:3.7-3.7` would give you Python3.7 on Alpine3.7?
Does the distro_ver also support a `latest` tag (i.e. `balenalib/raspberrypi3-alpine-python:3.7-latest`) or does `latest` only work for lang_ver ?

Reply:
Yes to the first question, 3.7 for both.

```
docker run -it balenalib/raspberrypi3-alpine-python:3.7-3.7 bash
bash-4.4# python --version
Python 3.7.4
bash-4.4# cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.7.3
PRETTY_NAME="Alpine Linux v3.7"
HOME_URL="http://alpinelinux.org"
BUG_REPORT_URL="http://bugs.alpinelinux.org"
```

Latest also works with distro so your example as expected gives Python 3.7 and latest Alpine 3.10.3